lib/tilebag.rb

```diff
@@ -0,0 +1,37 @@
+require 'pry'
+require_relative 'scoring.rb'
+require_relative 'player.rb'
+
+module Scrabble
+  class TileBag
+    attr_reader :default_tiles, :tiles_remaining
+
+    def initialize
+      @default_tiles = {"A" => 9,
+        "B" => 2, "C" => 2,
+        "D" => 4, "E" => 12, "F" => 2,
```

Comment:
By this logic, you're as likely to draw an `e` as you are a `z`! It also means that you won't ever get more than one of the same letter, which could be bad if you need to draw the last two tiles from the bag and they're both `i`. However, the way you've set up your data structures, it would be difficult to do it any other way.

Instead, you might keep all the letters in the bag in an array. You could seed the array with code like:

```ruby
def initialize
  letters = {
    # the hash you had before
  }
  @tiles = []
  letters.each do |letter, count|
    count.times do
      @tiles << letter
    end
  end
end
```

Then you could pull a random tile with `sample` and get a nice even distribution.